parrot/examples_test.go

```diff
@@ -0,0 +1,191 @@
+package parrot_test
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/rs/zerolog"
+	"github.com/smartcontractkit/chainlink-testing-framework/parrot"
+)
+
+func ExampleRegister() {
+	// Create a new parrot instance with no logging
+	p, err := parrot.Wake(parrot.WithLogLevel(zerolog.NoLevel))
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		err = p.Shutdown(context.Background())
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	// Create a new route /test that will return a 200 status code with a text/plain response body of "Squawk"
+	route := &parrot.Route{
+		Method:             http.MethodGet,
+		Path:               "/test",
+		RawResponseBody:    "Squawk",
+		ResponseStatusCode: 200,
+	}
+
+	// Register the route with the parrot instance
+	err = p.Register(route)
+	if err != nil {
+		panic(err)
+	}
+
+	// Call the route
+	resp, err := p.Call(http.MethodGet, "/test")
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	fmt.Println(resp.StatusCode)
+	body, _ := io.ReadAll(resp.Body)
+	fmt.Println(string(body))
+	// Output:
+	// 200
+	// Squawk
+}
+
+func ExampleRoute() {
```

Comment:
Maybe let's do an example with NodeSet and some blockchain to show how to mock EAs? Can be done later, though
https://github.com/smartcontractkit/chainlink-testing-framework/blob/main/framework/examples/myproject/smoke_test.go#L24

Reply:
Agreed it would be a helpful example, but I think it's a bit out of scope for just this tool. We should include examples like this as part of a larger, more comprehensive setup.